dissect/target/plugins/os/unix/log/messages.py

```diff
@@ -48,68 +42,65 @@ def messages(self):
-
-                        # Line can be a byte object on debian and centos
-                        if isinstance(line, bytes):
-                            line = line.decode("utf-8").strip()
-                        else:
-                            line = line.strip()
-
-                        if not line:
-                            continue
-
-                        try:
-                            ts = datetime.datetime.strptime(RE_TS.search(line).groups()[0], "%b %d %H:%M:%S")
-                            ts = ts.replace(year=year)
-                        except Exception:
-                            # Set ts to epoch 1970-01-01 if we could not convert
-                            ts = datetime.datetime.utcfromtimestamp(0)
-
-                        try:
-                            daemon = RE_DAEMON.search(line).groups()[0]
-                        except Exception:
-                            daemon = None
-
-                        try:
-                            pid = RE_PID.search(line)
-                            if pid is not None:
-                                pid = pid.groups()[0]
-                        except Exception:
-                            pid = None
-
-                        try:
-                            msg = RE_MSG.search(line).groups()[0]
-                            if msg is None:
-                                msg = line
-                        except Exception:
-                            msg = line
-
-                        yield MessagesRecord(
-                            ts=ts,
-                            filepath=f.path,
-                            daemon=daemon,
-                            pid=pid,
-                            msg=msg,
-                            raw=line,
-                            _target=self.target,
-                        )
+
+        log_files: [TargetPath] = list(self.target.fs.path("/var/log/").glob("syslog*")) + list(
+            self.target.fs.path("/var/log/").glob("messages*")
+        )
+
+        for log_file in log_files:
+
+            file_ctime = self.target.fs.get(str(log_file)).stat().st_ctime
+            year_file_created = datetime.fromtimestamp(file_ctime).year
+            last_seen_year = year_file_created
+            last_seen_month = 0
+
+            for line in decompress_and_readlines(log_file):
+
+                # Line can be a byte object on debian and centos
+                if isinstance(line, bytes):
+                    line = line.decode("utf-8").strip()
+                else:
+                    line = line.strip()
+
+                if not line or line == "":
+                    continue
+
+                try:
+                    ts = datetime.strptime(RE_TS.search(line).groups()[0], "%b %d %H:%M:%S")
+                    if last_seen_month > ts.month:
+                        last_seen_year += 1
+                    last_seen_month = ts.month
+                    ts = ts.replace(year=last_seen_year)
```

Comment:
I believe these timestamps are in local timezone? Preferably try to use the new locale plugin to get the correct timezone:
https://docs.python.org/3/library/datetime.html#timezone-objects

You can add the timezone using `ts.replace(year=last_seen_year, tzinfo=<timezone object here>)`

You can look at the Windows `datetime` plugin for inspiration:
https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugins/os/windows/datetime.py

Reply:
Are you still looking into this, since you left it unresolved? Or will that be a future fix?

Reply:
Currently implementing that suggestion, see the first example in https://github.com/fox-it/dissect.target/pull/127/commits/cafcdde77d62e865eb5994c53a3913d8a04ac607.

Reply:
Added in [6ad6518](https://github.com/fox-it/dissect.target/pull/131/commits/6ad6518d6a8b67b5ffaf0bd14c2ab58d612d6b12).